eventrecorder/recorder.go

```diff
@@ -147,99 +151,128 @@ func (r *EventRecorder) handleRetrievalEvents(res http.ResponseWriter, req *http
 		logger.Errorw("At least one retrieval event insertion failed", "err", err)
 		return
 	}
+	for _, event := range batch.Events {
+		recordMetrics(event)
+	}
```

Comment:
* Why a separate loop instead of doing this in the existing loop above?
* Double checking `recordMetrics` will never read the DB and would only interact with metrics. If we ever envision that it would do DB reads as part of metric collection, then we should probably do metric reporting asyncrhonously.

Reply:
no db reads. Seperate loop cause I want to insure we write successfully to the db before metrics